App crashes when trying to decrypt file

**Describe the bug**
App is crashing when I try to open an encrypted file directly (by pressing on it, can be tried with images).
Edit: bug is there for opening any file not just encrypted file, on my device.

**To Reproduce**
Steps to reproduce the behavior:
1. Encrypte an image
2. After encrypted, press on encrypted file and fill password.
3. Let it decrypt, openFile will throw NPE

**Expected behavior**
File should successfully open

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Smartphone (please complete the following information):**
 - Device: Oneplus 6t
 - OS: Android 10
 - Rooted: No
 - Version: master

**Additional context**
Add any other context about the problem here.
Stacktrace 
```
java.lang.RuntimeException: Error receiving broadcast Intent { act=decrypt_broadcast flg=0x10 (has extras) } in com.amaze.filemanager.fragments.MainFragment$5@8fd63cb
        at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1575)
        at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(Unknown Source:2)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7643)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
     Caused by: java.lang.IllegalArgumentException: Couldn't find meta-data for provider with authority com.amaze.filemanager
        at androidx.core.content.FileProvider.parsePathStrategy(FileProvider.java:606)
        at androidx.core.content.FileProvider.getPathStrategy(FileProvider.java:579)
        at androidx.core.content.FileProvider.getUriForFile(FileProvider.java:417)
        at com.amaze.filemanager.utils.files.FileUtils.openWith(FileUtils.java:457)
        at com.amaze.filemanager.utils.files.FileUtils.openFile(FileUtils.java:630)
        at com.amaze.filemanager.fragments.MainFragment$5.onReceive(MainFragment.java:832)
        at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1560)
```

--- a/app/src/main/java/com/amaze/filemanager/utils/files/FileUtils.java
+++ b/app/src/main/java/com/amaze/filemanager/utils/files/FileUtils.java
@@ -26,28 +26,25 @@ import android.animation.Animator;
 import android.animation.AnimatorListenerAdapter;
 import android.app.Activity;
 import android.content.ActivityNotFoundException;
-import android.content.ContentResolver;
 import android.content.Context;
 import android.content.Intent;
 import android.content.SharedPreferences;
 import android.content.pm.PackageManager;
 import android.content.pm.ResolveInfo;
-import android.database.Cursor;
 import android.media.MediaScannerConnection;
 import android.net.Uri;
 import android.os.AsyncTask;
 import android.os.Build;
 import android.os.CountDownTimer;
 import android.preference.PreferenceManager;
-import android.provider.BaseColumns;
-import android.provider.MediaStore;
-import android.text.TextUtils;
 import android.util.Log;
 import android.view.View;
 import android.widget.Toast;
+
 import androidx.annotation.NonNull;
 import androidx.core.content.FileProvider;
 import androidx.documentfile.provider.DocumentFile;
+
 import com.afollestad.materialdialogs.MaterialDialog;
 import com.amaze.filemanager.R;
 import com.amaze.filemanager.activities.DatabaseViewerActivity;
@@ -75,7 +72,7 @@ import com.cloudrail.si.interfaces.CloudStorage;
 import com.cloudrail.si.types.CloudMetaData;
 import com.googlecode.concurrenttrees.radix.ConcurrentRadixTree;
 import com.googlecode.concurrenttrees.radix.node.concrete.voidvalue.VoidValue;
-import jcifs.smb.SmbFile;
+
 import net.schmizz.sshj.sftp.RemoteResourceInfo;
 import net.schmizz.sshj.sftp.SFTPClient;
 import net.schmizz.sshj.sftp.SFTPException;
@@ -89,6 +86,8 @@ import java.util.Date;
 import java.util.LinkedList;
 import java.util.concurrent.atomic.AtomicLong;
 
+import jcifs.smb.SmbFile;
+
 /**
  * Functions that deal with files
  */
@@ -96,7 +95,6 @@ public class FileUtils {
 
     private static final String TAG = FileUtils.class.getSimpleName();
 
-    public static final String FILE_PROVIDER_AUTHORITY = "com.amaze.filemanager";
     public static final String FILE_PROVIDER_PREFIX = "storage_root";
     public static final String NOMEDIA_FILE = ".nomedia";
 
@@ -358,7 +356,7 @@ public class FileUtils {
         ArrayList<Uri> uris = new ArrayList<>();
         boolean b = true;
         for (File f : a) {
-            uris.add(FileProvider.getUriForFile(c, FILE_PROVIDER_AUTHORITY, f));
+            uris.add(FileProvider.getUriForFile(c, c.getPackageName(), f));
         }
 
         String mime = MimeTypes.getMimeType(a.get(0).getPath(), a.get(0).isDirectory());
@@ -398,7 +396,7 @@ public class FileUtils {
         String type = "application/vnd.android.package-archive";
 
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            Uri downloadedApk = FileProvider.getUriForFile(permissionsActivity.getApplicationContext(), "com.amaze.filemanager", f);
+            Uri downloadedApk = FileProvider.getUriForFile(permissionsActivity.getApplicationContext(), permissionsActivity.getPackageName(), f);
             intent.setDataAndType(downloadedApk, type);
             intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
         } else {
@@ -454,7 +452,7 @@ public class FileUtils {
      * Method supports showing a UI to ask user to open a file without any extension/mime
      */
     public static void openWith(final File f, final Context c, final boolean useNewStack) {
-        openWithInternal(FileProvider.getUriForFile(c, FILE_PROVIDER_AUTHORITY, f), c, useNewStack);
+        openWithInternal(FileProvider.getUriForFile(c, c.getPackageName(), f), c, useNewStack);
     }
 
     public static void openWith(final DocumentFile f, final Context c, final boolean useNewStack) {
@@ -624,7 +622,7 @@ public class FileUtils {
                 m.startActivity(intent);
         } else {
             try {
-                openUnknownInternal(FileProvider.getUriForFile(m, FILE_PROVIDER_AUTHORITY, f), MimeTypes.getMimeType(f.getAbsolutePath(), false), m, false, useNewStack);
+                openUnknownInternal(FileProvider.getUriForFile(m, m.getPackageName(), f), MimeTypes.getMimeType(f.getAbsolutePath(), false), m, false, useNewStack);
             } catch (Exception e) {
                 Toast.makeText(m, m.getString(R.string.no_app_found),Toast.LENGTH_LONG).show();
                 openWith(f, m, useNewStack);

